### PR TITLE
(PT-318) Update apply_helpers to latest version of bolt

### PR DESCRIPTION
**Pull request by donoghuc**
This includes making apply/apply_prep work for remote targets and supporting the get_resources plan function.

### CODE REVIEW

File: tasks/query_resources.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  ],
+  "parameters": {
+    "plugins": {
+      "description": "Custom facts bundle to use when applying resources",
+      "type": "String",
+      "sensitive": true
+    },
+    "resources": {
+      "description": " A resource type or instance, or an array of such",
+      "type": "Variant[String, Type[Resource], Array[Variant[String, Type[Resource]]]]"
```
Comment (ekinanp) [inline tasks/query_resources.json]: Hmm, why not simplify this to `Array[Variant[String, Type[Resource]]]`? I also don't think the existing code would work with a non-array argument b/c of the `flat_map` call.

Comment (donoghuc) [reply tasks/query_resources.json]: ah, good call. yeah looks like the plan function ensures there is a flatt array of that type. Will update thanks. https://github.com/puppetlabs/bolt/blob/a9ae282d31d993b27befa8c4f22ffc7d90a95037/bolt-modules/boltlib/lib/puppet/functions/get_resources.rb#L22

**Review by ekinanp — APPROVED** [review]

File: tasks/query_resources.json (hunk 1 above)
Comment (nicklewis) [reply tasks/query_resources.json]: It's actually just `Array[String]`. The function stringifies them all. https://github.com/puppetlabs/bolt/blob/a9ae282d31d993b27befa8c4f22ffc7d90a95037/bolt-modules/boltlib/lib/puppet/functions/get_resources.rb#L53

`Type[Resource]` isn't `Data` so you can't pass it as a JSON task parameter. I actually thought there would be an error somewhere if the specified type wasn't a subset of `Data` but maybe not.

Comment (donoghuc) [reply tasks/query_resources.json]: aww yeah, I will get a pr up to fix that. My bad

**Pull request merged**